universal-login-sdk/lib/sdk.ts

```diff
@@ -127,18 +127,34 @@ class UniversalLoginSDK {
     return this.relayerApi.getConfig();
   }
 
+  async waitForStatus(messageId: string, timeout: number = 1000, tick: number = 20) {
+    let elapsed = 0;
+    let status;
+    do {
+      if (elapsed > timeout) {
+        throw Error('Timeout');
+      }
+      await sleep(tick);
+      elapsed += tick;
+      status =  await this.relayerApi.getStatusById(messageId);
+    } while (status && !status.transactionHash && !status.error);
+    if (status.error) {
+      throw Error(status.error);
+    }
+    return status;
+  }
+
   async execute(message: Message, privateKey: string) {
     const unsignedMessage = {
       ...this.defaultPaymentOptions,
       ...message,
       nonce: message.nonce || parseInt(await this.getNonce(message.from!, privateKey), 10),
     } as MessageWithFrom;
     const signedMessage = await createSignedMessage(unsignedMessage, privateKey);
-
-    const result = await this.relayerApi.execute(signedMessage);
-
-    if (result.transaction.hash) {
-      const transactionHash = result.transaction.hash;
+    const result = await this.relayerApi.execute(stringifySignedMessageFields(signedMessage));
+    if (typeof result.transaction === 'number') {
```

Comment:
Why would `result.transaction` type be number?

Reply:
Changed ✔️